baybe/acquisition/acqfs.py

```diff
@@ -1,11 +1,106 @@
 """Available acquisition functions."""
 
+import math
 from typing import ClassVar
 
+import pandas as pd
+from attr.validators import optional as optional_v
 from attrs import define, field
-from attrs.validators import ge, instance_of
+from attrs.validators import ge, gt, instance_of, le
 
 from baybe.acquisition.base import AcquisitionFunction
+from baybe.searchspace import SearchSpace
+from baybe.utils.sampling_algorithms import (
+    SamplingMethod,
+    sample_numerical_df,
+)
+
+
+########################################################################################
+### Active Learning
+@define(frozen=True)
+class qNegIntegratedPosteriorVariance(AcquisitionFunction):
+    """Monte Carlo based negative integrated posterior variance.
+
+    This is particularly useful for active learning as it is a measure for global model
+    uncertainty.
+    """
+
+    abbreviation: ClassVar[str] = "qNIPV"
+
+    sampling_fraction: float = field(
+        converter=float,
+        validator=[gt(0.0), le(1.0)],
+        default=1.0,
+    )
+    """Fraction of data that will be sampled for integrating the posterior.
+
+    The fraction will be ignored if 'sampling_n_points' is not None."""
+
+    sampling_n_points: int | None = field(
+        validator=optional_v([gt(0), instance_of(int)]),
+        default=None,
+    )
+    """Number of data points that will be sampled for integrating the posterior."""
+
+    sampling_method: SamplingMethod = field(
+        converter=SamplingMethod, default=SamplingMethod.Random
+    )
+    """Strategy used for sampling data for integrating the posterior."""
+
+    def get_integration_points(self, searchspace: SearchSpace) -> pd.DataFrame:
+        """Sample points from a search space for integration purposes.
+
+        Sampling of the discrete part can be controlled via 'sampling_method', but
+        sampling of the continuous part will always be random.
+
+        Args:
+            searchspace: The searchspace from which to sample integration points.
+
+        Returns:
+            The sampled data points.
+
+        Raises:
+            ValueError: If searchspace is purely continuous and 'sampling_n_points' was
+                not provided.
+        """
+        sampled_parts = []
+        n_candidates = None
```

Comment:
Just to help the reader a little bit (and perhaps support mypy in finding bugs in case of future changes)
```suggestion
        sampled_parts: list[pd.DataFrame] = []
        n_candidates: int | None = None
```